agent/bootstrap.go

```diff
@@ -121,14 +125,30 @@ func (boot bootstrap) setUserPasswords(env boshsettings.Env) error {
 		return nil
 	}
 
+	if !env.GetKeepRootPassword() {
+		err := boot.platform.SetUserPassword(boshsettings.RootUsername, password)
+		if err != nil {
+			return bosherr.WrapError(err, "Setting root password")
+		}
+	}
+
 	err := boot.platform.SetUserPassword(boshsettings.RootUsername, password)
 	if err != nil {
-		return bosherr.WrapError(err, "Setting root password")
+		return bosherr.WrapError(err, "Setting vcap password")
+	}
+
+	return nil
+}
+
+func (boot bootstrap) setRootPasswords(env boshsettings.Env) error {
```

Comment:
After reading this code it doesnt look like it properly works for vcap users anymore. I think something like this would be easier to understand:

```
func (boot bootstrap) setUserPasswords(env boshsettings.Env) error {
	password := env.GetPassword()
	if password == "" {
		return nil
	}

	err := boot.platform.SetUserPassword(boshsettings.VCAPUsername, password)
	if err != nil {
		return bosherr.WrapError(err, "Setting vcap password")
	}

	if !env.KeepRootPassword() {
		err := boot.platform.SetUserPassword(boshsettings.RootUsername, password)
		if err != nil {
			return bosherr.WrapError(err, "Setting root password")
		}
	}

	return nil
}
```

and remove SetRootPasswords